sunpy/net/helio/hec.py

```diff
@@ -173,9 +175,9 @@ def search(self, *args, **kwargs):
                                                     MAXRECORDS=max_records)
         results = votable_handler(etree.tostring(results))
         table = HECResponse(results.to_table(), client=self)
-        if len(table) == max_records == 500:
-            warn_user("Number of results is the same as the default `max_records` of 500. "
-                      "It is possible your query has been truncated. "
+        if len(table) == max_records:
+            warn_user("Number of results is the same as the " + ("default " if max_records == 500 else "") + "`a.helio.MaxRecords` limit of " + str(max_records) +
+                      ". It is possible your query has been truncated. "
```

Comment:
```suggestion
            warn_user("Number of results is the same as current limit. "
                      "It is possible your query has been truncated. "
```

How about just?

Reply:
Yes, but I was thinking about showing more information about the source of the warning.

Reply:
I think it's confusing currently?

Reply:
This line will also need a unit test